internal/konnect/node_agent_test.go

```diff
@@ -381,3 +382,82 @@ func TestNodeAgent_StartDoesntReturnUntilContextGetsCancelled(t *testing.T) {
 	case <-agentReturned:
 	}
 }
+
+func TestNodeAgent_ControllerNodeStatusGetsUpdatedOnStatusNotification(t *testing.T) {
+	nodeClient := newMockNodeClient(nil)
+	configStatusQueue := newMockConfigStatusNotifier()
+	gatewayClientsChangesNotifier := newMockGatewayClientsNotifier()
+
+	nodeAgent := konnect.NewNodeAgent(
+		testHostname,
+		testKicVersion,
+		konnect.DefaultRefreshNodePeriod,
+		logr.Discard(),
+		nodeClient,
+		configStatusQueue,
+		newMockGatewayInstanceGetter(nil),
+		gatewayClientsChangesNotifier,
+		newMockManagerInstanceIDProvider(uuid.New()),
+	)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	agentReturned := make(chan struct{})
+	go func() {
+		require.NoError(t, nodeAgent.Start(ctx))
+		close(agentReturned)
+	}()
+
+	testCases := []struct {
+		notifiedConfigStatus    clients.ConfigStatus
+		expectedControllerState nodes.IngressControllerState
+	}{
+		{
+			notifiedConfigStatus:    clients.ConfigStatusOK,
+			expectedControllerState: nodes.IngressControllerStateOperational,
+		},
+		{
+			notifiedConfigStatus:    clients.ConfigStatusTranslationErrorHappened,
+			expectedControllerState: nodes.IngressControllerStatePartialConfigFail,
+		},
+		{
+			notifiedConfigStatus:    clients.ConfigStatusApplyFailed,
+			expectedControllerState: nodes.IngressControllerStateInoperable,
+		},
+		{
+			notifiedConfigStatus:    clients.ConfigStatusOKKonnectApplyFailed,
+			expectedControllerState: nodes.IngressControllerStateOperationalKonnectOutOfSync,
+		},
+		{
+			notifiedConfigStatus:    clients.ConfigStatusTranslationErrorHappenedKonnectApplyFailed,
+			expectedControllerState: nodes.IngressControllerStatePartialConfigFailKonnectOutOfSync,
+		},
+		{
+			notifiedConfigStatus:    clients.ConfigStatusApplyFailedKonnectApplyFailed,
+			expectedControllerState: nodes.IngressControllerStateInoperableKonnectOutOfSync,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprint(tc.notifiedConfigStatus), func(t *testing.T) {
+			configStatusQueue.NotifyConfigStatus(ctx, tc.notifiedConfigStatus)
+
+			require.Eventually(t, func() bool {
+				controllerNode, ok := lo.Find(nodeClient.MustAllNodes(), func(n *nodes.NodeItem) bool {
+					return n.Type == nodes.NodeTypeIngressController
+				})
+				if !ok {
+					t.Log("controller node not found")
+					return false
+				}
+
+				if controllerNode.Status != string(tc.expectedControllerState) {
+					t.Logf("expected controller node status to be %q, got %q", tc.expectedControllerState, controllerNode.Status)
+					return false
+				}
+
+				return true
+			}, time.Second, time.Millisecond*10)
```

Comment:
```suggestion
			}, time.Second, time.Millisecond)
```

Be aggressive! What we want? Short tests!